padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,216 @@
+            activation_function(),
+            torch.nn.ConvTranspose2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size,
+                                     stride=stride),
+        )
+        self.linear_3 = torch.nn.Linear(in_features=1, out_features=self.size_bottleneck**2)
+
+    def encode(self, x):
+        x = self.conv_encoder(x)
+        x = rearrange(x, 'b c x y -> b (c x y)')
+        mu = self.linear_1(x)  # mean value
+        std = 0.5 * self.linear_2(x)  # standard deviation
+        eps = torch.randn_like(std)
+        return mu + eps*std, mu, std
+
+    def decode(self, x):
+        x = self.linear_3(x)
+        x = rearrange(x, 'b (x y) -> b x y', y=self.size_bottleneck, x=self.size_bottleneck)
+        x = x[:, None, :, :]
+        return self.transp_conv_decoder(x)
+
+    def forward(self, input):
+        x, mu, std = self.encode(input)
+        y = self.decode(x)
+        return y[:, :, :28, :28], mu, std
+
+
+@ex.config
+def config():
+    epochs = 2          # Use resume=True to train for more epochs
+    batch_size = 10
+    resume = False      # PT: Continue from checkpoints
+
+    assert os.environ.get('STORAGE_ROOT') is not None, 'STORAGE_ROOT is not defined in the environment variables'
+    logfilepath = os.environ['STORAGE_ROOT'] + '/vae_mnist'  # PT creates log files
+
+    activation_function = ACTIVATION_FN_MAP['relu']  # activation function after convolutional layers
+    hidden_channels = 64
+    kernel_size = 2
+    assert kernel_size < 6, 'maximum kernel_size is 5'
+
+
+@ex.automain
+def main(activation_function, batch_size, epochs, hidden_channels, kernel_size, resume, logfilepath):
+    """
+    Example Training for a Variational Autoencoder uses SGD for optimation
+
+    Args:
+        epochs: number of epochs, use resume=True to train for more epochs
+        batch_size: number of batches
+        resume: PT: Continue from checkpoints
+        logfilepath: PT creates log files
+                    The structure of produced storage_dir is:
+                    .
+                    ├── checkpoints
+                    │   ├── ckpt_7122.pth
+                    │   ├── ckpt_14244.pth
+                    │   ├── ckpt_best_loss.pth -> ckpt_7122.pth
+                    │   ├── ckpt_latest.pth -> ckpt_14244.pth
+                    │   └── ckpt_ranking.json
+                    ├── events.out.tfevents.1548851867.ntsim5
+    """
+    # load dataset
+    train_set, test_set = get_dataset(batch_size=batch_size)
+
+    # load model
+    model = VAEModel(activation_function, hidden_channels, kernel_size)
+
+    trainer = pt.trainer.Trainer(
+        model=model,
+        storage_dir=logfilepath,
+        optimizer=pt_opt.SGD(),
+        stop_trigger=(epochs, 'epoch'),
+        #checkpoint_trigger=(5000, 'iteration'),
+        summary_trigger=(1000, 'iteration'),
+    )
+
+    trainer.register_validation_hook(validation_iterator=test_set)
+    trainer.train(train_set, resume=resume)
```

Comment:
To show that the model is actually working, it would be cool if you finally generate some new images by decoding some z's sampled from a standard normal distribution and save them in logfilepath.

Reply:
We could add those images to a README.md to show what the user can expect when training this model